control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -287,12 +377,18 @@ func (r *AWSEndpointServiceReconciler) Reconcile(ctx context.Context, req ctrl.R
 			// If we previously removed our finalizer, don't delete again and return early
 			return ctrl.Result{}, nil
 		}
-		completed, err := r.delete(ctx, awsEndpointService, r.ec2Client, r.route53Client)
+
+		ec2Client, route53Client, err := r.awsClientBuilder.getClients()
 		if err != nil {
-			return ctrl.Result{}, fmt.Errorf("failed to delete resource: %w", err)
-		}
-		if !completed {
-			return ctrl.Result{RequeueAfter: endpointServiceDeletionRequeueDuration}, nil
+			log.Error(err, "failed to get AWS client, skipping aws endpoint service cleanup")
```

Comment:
does this introduce a new scenario that wasn't possible before? we remove the finalizer if say a transient issue fails to get the client?

Reply:
The only 2 possible errors are that the session cannot be created because env/cfg is incomplete or that the client builder was not initialized from a hcp. Either case is not recoverable with a retry.